io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java

```diff
@@ -65,6 +65,9 @@ public void sendForLogin(HttpServletRequest request, HttpServletResponse respons
         RequestContext requestContext = defaultOIDCManager.sendForLogin(request, response);
         if (request.getRequestURI() != null) {
             String redirectPageURI = request.getRequestURI().substring(request.getContextPath().length() + 1);
+            if (request.getQueryString() != null) {
+                redirectPageURI = redirectPageURI + "?" + request.getQueryString();
```

Comment:
let's use a StringBuilder to build the redirect URI

Reply:
Fixed